# Tasks                          

## Task 1: Source the data set

# question -- Import the Iris data set from the sklearn.datasets module

In [35]:



# firstly import the pandas function to read comma seperated values (csv), which will be in the data set
# rename to pd for short

import pandas as pd

# import the sklearn package which is a machine learning libraries for Python.good for practicing on datasets.
# rename to skl for short

import sklearn as skl

#load the iris dataset
#df is a pandas fuction that will display dataframe

df=pd.read_csv('https://gist.githubusercontent.com/curran/a08a1080b88344b0c8a7/raw/0e7a9b0a5d22642a06d3d5b9bcbad9890c8ee534/iris.csv')

# df to display
df


#Reference: pandas- "https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html"
#Reference: Ian McLoughlin Lecture : week 3 practical-pands and loading datasets

sepal_length  sepal_width  petal_length  petal_width    species
0             5.1          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[150 rows x 5 columns]

# question:  Explain, in your own words, what the load_iris() function returns.



load_iris() is a panda function that enables you to load in data from Scikit-learn , and you can then chose what specific data you would like to be displayed like in the example below with "feature_names"

"load_iris() is a a quick way to load the popular iris dataset from Scikit-learn and it is used to practicing classification algorithms and exploratory data analysis."
Reference: chatgpt- prompt-explain load_iris(), "https://chatgpt.com/c/6813ac34-43f0-8003-b08c-170c373f4fb8"

Reference: Ian McLoughlin Lecture : week 3 practical-pands and loading datasets

In [36]:
import pandas as pd
import sklearn as skl
from sklearn.datasets import load_iris

#how to display the dataset imported-
#rename function to iris

iris = load_iris()

#example of displaying information from set like feature names

print(iris.feature_names)    

['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


## Task 2: Explore the data structure.

## question -Print and explain the shape of the data set, the first and last 5 rows of the data, the feature names, and the target classes.

In [37]:
# to do this i will need to use other panda functions
# all which can be found on the pandas website- Reference: "https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html"
# also Chatgpt:prompt- to get the last five rows, "https://chatgpt.com/c/6813ac34-43f0-8003-b08c-170c373f4fb8"

# for the last 5 rows use : print to display, df for dataframe, and for the last 5 rows - .tail() function

print(df.tail())



     sepal_length  sepal_width  petal_length  petal_width    species
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica


explaining the data shown- in last 5 rows as shown above the function is displaying the last 5 rows of data from specimen 145 to 149 , aslo includes the column for values in length, width for sepal and length and width for petal , also the species column

same as above but for the feature name, also from the same sources- iris.feature_names function

In [38]:
iris.feature_names

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

explaining the data shown- sepal length and width and petal length and width in centimetres are the feature names 

same as above but for the target classes, also from the same sources-iris.target_names function

In [39]:
iris.target_names

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

explaining the data shown- array is all the target names(classes) in the dataset which in this case are setosa,versicolor and virginica 

## Task 3: summarize the data

## Question -  For each feature in the dataset, calculate and display:

- mean
-minimum
-maximum
-standard deviation
-median


"NumPy (short for Numerical Python) is a popular open-source Python library that's mainly used for working with numerical data. It provides powerful tools for handling large, multi-dimensional arrays and matrices, along with a huge collection of mathematical functions to operate on these arrays efficiently." Reference chatgpt- prompt, what is numPy, "https://chatgpt.com/c/6813c8eb-44bc-8003-94bb-968f3aa2c8dc

firstly import numPy as we are working with large array of data , will also rename to np for short:

In [ ]:



import numpy as np

# for mean use numPy mean() function

# all information gathered for this question referenced below
# reference: chatgpt: prompt -how to get the mean of the iris dataset, "https://chatgpt.com/c/6813d08b-691c-8003-842d-96eadc100046".
# reference: W3schools (2025), machine learning-mean, median and mode [online] 
# Available at: https://www.w3schools.com/python/python_ml_mean_median_mode.asp. [accessed 01 may 2025].
# Reference: geeksforgeeks.com (2025) Pandas Functions in Python: A Toolkit for Data Analysis. [online]
# Available from: https://www.geeksforgeeks.org/pandas-functions-in-python/ [accessed 1 may 2025].


df = pd.DataFrame(iris.data, columns=iris.feature_names)

# apply
mean_values = df.mean()

#display
print(mean_values)

sepal length (cm)    5.843333
sepal width (cm)     3.057333
petal length (cm)    3.758000
petal width (cm)     1.199333
dtype: float64


In [ ]:
# for minimum value: use .min() function pandas

#apply
min_values = df.min()


#display
print("Minimum values:", min_values)


Minimum values: sepal length (cm)    4.3
sepal width (cm)     2.0
petal length (cm)    1.0
petal width (cm)     0.1
dtype: float64


In [ ]:
# for maximum value: .max() function pandas

#apply
max_values = df.max()

#display
print("Maximum values:", max_values)

Maximum values: sepal length (cm)    7.9
sepal width (cm)     4.4
petal length (cm)    6.9
petal width (cm)     2.5
dtype: float64


In [ ]:
# for standard deviation: .std() function pandas

#apply
std_values = df.std()

#display
print("Standard Deviation:", std_values)

Standard Deviation: sepal length (cm)    0.828066
sepal width (cm)     0.435866
petal length (cm)    1.765298
petal width (cm)     0.762238
dtype: float64


In [48]:
# for median :.median() function pandas

#apply
median_values = df.median()

#display
print("Median values:", median_values)

Median values: sepal length (cm)    5.80
sepal width (cm)     3.00
petal length (cm)    4.35
petal width (cm)     1.30
dtype: float64


## Task 4. Visualize Features

 Plot histograms for each feature using matplotlib.
 Add appropriate titles and axis labels